### PR TITLE
adding catch all logging for exception during DQL/DML process

### PR DESCRIPTION
**Pull request by walterddr**
previously b/c `BrokerRequestHandler#handleRequest` can throw exception. it is possible that there's only exception being and logged without the context such as what request content is being processed.

adding a catch, log then rethrow logic in the client side

### CODE REVIEW

File: pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java [hunk 1]
```diff
@@ -316,12 +316,22 @@ private BrokerResponse executeSqlQuery(ObjectNode sqlRequestJson, HttpRequesterI
...
-        httpRequesterIdentity.getHttpHeaders().entries()
-            .forEach(entry -> headers.put(entry.getKey(), entry.getValue()));
-        return _sqlQueryExecutor.executeDMLStatement(sqlNodeAndOptions, headers);
+        try {
+          Map<String, String> headers = new HashMap<>();
+          httpRequesterIdentity.getHttpHeaders().entries()
+              .forEach(entry -> headers.put(entry.getKey(), entry.getValue()));
+          return _sqlQueryExecutor.executeDMLStatement(sqlNodeAndOptions, headers);
+        } catch (Exception e) {
+          LOGGER.error("Error handling DQL request:\n{}\nException: {}", sqlRequestJson,
```
Comment (xiangfu0) [inline pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:331]: DML

**Review by xiangfu0 — APPROVED** [review]

File: pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java (hunk 1 above)
Comment (walterddr) [reply pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:331]:
```suggestion
          LOGGER.error("Error handling DML request:\n{}\nException: {}", sqlRequestJson,
```

File: pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java [hunk 2]
```diff
@@ -316,12 +316,22 @@ private BrokerResponse executeSqlQuery(ObjectNode sqlRequestJson, HttpRequesterI
         try (RequestScope requestStatistics = Tracing.getTracer().createRequestScope()) {
           return _requestHandler.handleRequest(sqlRequestJson, sqlNodeAndOptions, httpRequesterIdentity,
               requestStatistics, httpHeaders);
+        } catch (Exception e) {
+          LOGGER.error("Error handling DQL request:\n{}\nException: {}", sqlRequestJson,
+              QueryException.getTruncatedStackTrace(e));
+          throw e;
```
Comment (walterddr) [inline pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:322]: @xiangfu0 any idea what kind of error we should wrap and rethrow here? simply pass through seems a bit less informative than I thought of but i couldn't think of a suitable one

Comment (xiangfu0) [reply pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:322]: I would say, here should just be 500 internal error with some truncated stacktrace

Comment (walterddr) [reply pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:322]: already merged. i think we can improve later (this PR only rethrow and doesn't change the behavior)

**Pull request merged**